File: backend/src/contexts/backoffice/movies/application/movie_searcher.py
from src.contexts.backoffice.movies.domain.movie import Movie
from src.contexts.backoffice.movies.domain.movie_repository import MovieRepository
from src.contexts.shared.domain.criteria import Criteria


class MovieSearcher:
    def __init__(self, repository: MovieRepository) -> None:
        self.repository = repository

    def run(self, criteria: dict | None = None) -> list[Movie]:
        if criteria is None:
            return self.repository.search_all()
        return self.repository.matching(Criteria.from_primitives(**criteria))

File: frontend/src/contexts/backoffice/movies/application/MovieSearcher.ts
import { Movie } from "@/contexts/backoffice/movies/domain/Movie";
import { MovieRepository, PaginatedResponse } from "@/contexts/backoffice/movies/domain/MovieRepository";

export class MovieSearcher {
    private readonly movieRepository: MovieRepository;

    constructor(movieRepository: MovieRepository) {
        this.movieRepository = movieRepository;
    }

    async run(): Promise<PaginatedResponse<Movie>> {
        return await this.movieRepository.matching();
    }
}
